### PR TITLE
257 bug preprocessing inspector does not show correct x axis when feature selection used in pipeline

### LINKED ISSUE #257: bug: Preprocessing inspector does not show correct  x_axis when feature selection used in pipeline
**Describe the bug**
The `PreprocessingInspector` loses the original semantic `x_axis` labels (such as wavenumbers) whenever a feature selection step (e.g., `RangeCut`, `IndexSelector`) is included in the pipeline. After the feature selection step modifies the dimensionality of the data, all subsequent step plots in `.inspect()` and the preprocessed spectra plot in `.inspect_spectra()` fall back to displaying generic integer indices (`0, 1, 2, ...`) instead of the correct subset of the original axis labels. 

This behavior is inconsistent with `PCAInspector` and `PLSRegressionInspector`, which correctly preserve and filter the `x_axis` labels using the `get_support()` masks from the pipeline's `SelectorMixin` steps.

**To Reproduce**
1. Create a scikit-learn pipeline that includes a feature selection step (e.g., `RangeCut`) followed by another preprocessing step.
2. Fit the pipeline on some training data `X_train`.
3. Initialize the `PreprocessingInspector` with the pipeline, datasets, and a custom `x_axis` array (e.g., wavenumbers).
4. Run `inspector.inspect()`.
5. Observe that the plot for the raw spectra correctly shows the provided `x_axis` labels, but any plot for steps *after* the `RangeCut` step will show integer indices on the X-axis.

**Expected behavior**
The `PreprocessingInspector` should track which features were selected by querying `SelectorMixin.get_support()` for any feature selection steps in the pipeline. The plots for all subsequent preprocessing steps (and the final preprocessed spectra) should display the correctly filtered original `x_axis` labels, maintaining exact consistency with the regression and PCA inspectors.

**Architectural Root Cause:** 
Currently, `PreprocessingInspector` extends `_DataHoldingBase` rather than `_BaseInspector`. `_BaseInspector` handles feature selection properly through `_get_feature_mask()` and `_get_preprocessed_feature_names()`. Because `PreprocessingInspector` bypasses `_BaseInspector`, it implemented a naive fallback in `_x_axis_for_n_features()` which strictly relies on checking if `n_features_out == self.n_features_in_`. When the lengths differ, it instantly reverts to `np.arange`.

**Proposed Fix Implementation:**
To resolve this correctly without code duplication, the solution refactors `x_axis` resolution out of `_BaseInspector` and into `_DataHoldingBase` (which actually owns the `x_axis` state). 
- A new method `_resolve_x_axis_after_transform(steps)` dynamically computes the correct output axis by sequentially applying the boolean masks of any `SelectorMixin` steps.
- Both `_BaseInspector` and `PreprocessingInspector` now delegate to this shared logic, ensuring a single source of truth for axis handling across the entire architecture.

### PR DESCRIPTION
**Pull request by paucablop**
Bug fixes for Processing Inspector:

- Correct display of `x_axis` when feature selector is available
- Correct display of discrete colormap

### CODE REVIEW

File: chemotools/inspector/helpers/_preprocessing.py [hunk 1]
```diff
@@ -61,7 +61,6 @@ def create_preprocessing_step_plot(
         x=x_axis,
         y=X,
         color_by=color_values,
-        colormap="shap",
         labels=empty_labels,
         color_mode=color_mode,
     )
```
[the remark this answers is not in the document]
Comment (paucablop) [reply chemotools/inspector/helpers/_preprocessing.py:66]: this is solved

File: chemotools/inspector/core/base.py [hunk 2]
```diff
@@ -431,13 +461,15 @@ def _get_preprocessed_feature_names(
...
         X = self._get_preprocessed_data(base_dataset)
-        return np.arange(X.shape[1])
+        if self.transformer_ is None:
+            steps = []
+        elif isinstance(self.transformer_, Pipeline):
+            steps = self.transformer_.steps
+        else:
+            steps = [("transformer", self.transformer_)]
+
+        return self._resolve_x_axis_after_transform(steps, X.shape[1])
```
[the remark this answers is not in the document]
Comment (paucablop) [reply chemotools/inspector/core/base.py:451]: Done
